### PR TITLE
feat(document-search) add location metadata to document

### PR DESCRIPTION
**Pull request by kdziedzic68**

### CODE REVIEW

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py [hunk 1]
```diff
@@ -50,6 +60,19 @@ def id(self) -> str:
         """
         return self.source.id
 
+    def add_location_metadata(self, provider_metadata: dict | None) -> None:
```
Comment (mhordynski) [inline packages/ragbits-document-search/src/ragbits/document_search/documents/document.py:63]: why `| None` here?

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py [hunk 2]
```diff
@@ -32,13 +32,23 @@ class DocumentType(str, Enum):
     UNKNOWN = "unknown"
 
 
+class DocumentLocation(BaseModel):
```
Comment (mhordynski) [inline packages/ragbits-document-search/src/ragbits/document_search/documents/document.py:35]: I'm wondering if that model applies for each document type that we support. For example for PDF files location may differ from MD location

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py (hunk 1 above)
Comment (kdziedzic68) [reply packages/ragbits-document-search/src/ragbits/document_search/documents/document.py:63]: in unstructured output format metadata was marked as optional AFAIR

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py (hunk 2 above)
Comment (kdziedzic68) [reply packages/ragbits-document-search/src/ragbits/document_search/documents/document.py:35]: the metadata for unstructured output has document agnostic format. have inspected it and did not see anything that could be document type dependent - would leave a comment there to potentially keep in mind for the future

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py [hunk 3]
```diff
@@ -32,13 +32,23 @@ class DocumentType(str, Enum):
...
+
+
 class DocumentMeta(BaseModel):
     """
     An object representing a document metadata.
     """
 
     document_type: DocumentType
     source: LocalFileSource | GCSSource | HuggingFaceSource = Field(..., discriminator="source_type")
+    location: DocumentLocation | None = None
```
Comment (micpst) [inline packages/ragbits-document-search/src/ragbits/document_search/documents/document.py:51]: I think this should be defined in the `Element` class

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py (hunk 1 above)
Comment (micpst) [reply packages/ragbits-document-search/src/ragbits/document_search/documents/document.py:63]: I don't see optional typing for `element.metadata` anywhere, but still you use this method with `model_dump` in the condition, so you're sure it's a dict here

File: packages/ragbits-document-search/src/ragbits/document_search/documents/document.py (hunk 3 above)
Comment (kdziedzic68) [reply packages/ragbits-document-search/src/ragbits/document_search/documents/document.py]: done

**Review by micpst — APPROVED** [review]

**Pull request merged**